src/query/join.rs

```diff
@@ -637,4 +637,292 @@ mod tests {
+                    Some(fruit_model(2, "orange".to_owned(), Some(1)))
+                ),
+                (
+                    cake_model(2, "orange cake".to_owned()),
+                    Some(fruit_model(2, "orange".to_owned(), Some(2)))
+                ),
+                (
+                    cake_model(3, "chocolate cake".to_owned()),
+                    None
+                )
+            ]
+        );
+
+        Ok(())
+    }
+
+    #[smol_potat::test]
+    #[cfg(any(
+        feature = "mock",
+    ))]
+    pub async fn with_related() -> Result<(), sea_orm::DbErr> {
+        use sea_orm::{MockDatabase, DbBackend, EntityTrait, QuerySelect, Transaction, Statement};
+        use sea_orm::tests_cfg::*;
+
+        let db = MockDatabase::new(DbBackend::Postgres)
+            .append_query_results([[
+                cake_fruit_model(1, "apple cake".to_owned(), 1, "apple".to_owned()),
+                cake_fruit_model(1, "apple cake".to_owned(), 2, "orange".to_owned())
+            ]])
+            .into_connection();
+
+        assert_eq!(
+            Cake::find()
+            .find_with_related(Fruit)
+            .select_only()
+            .column(cake::Column::Name)
+            .column(fruit::Column::Name)
+            .all(&db)
+            .await?,
+            [
+                (
+                    cake_model(1, "apple cake".to_owned()),
+                    vec![
+                        fruit_model(1, "apple".to_owned(), Some(1)), 
+                        fruit_model(2, "orange".to_owned(), Some(1))
+                    ]
+                )
+            ]
+        );
+
+        assert_eq!(
+            db.into_transaction_log(),
+            [Transaction::many([Statement::from_sql_and_values(
+                DbBackend::Postgres,
+                [
+                    r#"SELECT "cake"."name", "fruit"."name""#,
+                    r#"FROM "cake""#,
+                    r#"LEFT JOIN "fruit" ON "cake"."id" = "fruit"."cake_id""#,
+                    r#"ORDER BY "cake"."id" ASC"#
+                ]
+                .join(" ")
+                .as_str(),
+                []
+            ),])]
+        );
+
+        Ok(())
+    }
+
+    #[smol_potat::test]
+    #[cfg(any(
+        feature = "mock",
+    ))]
+    pub async fn with_related_2() -> Result<(), sea_orm::DbErr> {
+        use sea_orm::{MockDatabase, DbBackend, EntityTrait, QuerySelect};
+        use sea_orm::tests_cfg::*;
+
+        let db = MockDatabase::new(DbBackend::Postgres)
+            .append_query_results([[
+                cake_fruit_model(1, "apple cake".to_owned(), 1, "apple".to_owned()),
+                cake_fruit_model(2, "fruit cake".to_owned(), 1, "apple".to_owned()),
+                cake_fruit_model(2, "fruit cake".to_owned(), 2, "orange".to_owned()),
+                cake_fruit_model(2, "fruit cake".to_owned(), 3, "grape".to_owned()),
+                cake_fruit_model(2, "fruit cake".to_owned(), 4, "melon".to_owned()),
```

Comment:
May be?

```suggestion
                cake_only(3, "fruit cake".to_owned()),
```

Reply:
This is a problem to be fixed (same as the also_related_4) where I can't have mock_query_result of different types.
If I do it as suggested, it would give me error "expected (Model, Model) but found Model"